setup.py

```diff
@@ -15,7 +15,7 @@ def get_version() -> str:
 
 def get_install_requires() -> str:
     return [
-        "gym>=0.15.4",
+        "gym>=0.23.1",
```

Comment:
I personally suggest supporting both for compatibility. Otherwise this pull request must be blocked.

Reply:
I don't think this quite makes sense? Right now, there have been no large breaking changes made to 0.23.1, all the major stuff is new APIs. All the other RL libraries (RLlib, SB3, etc.) have or are about to use a minimum version of the current version of Gym leading up to the 1.0 release. Environments have largely been cleaned up to this end as well.

Reply:
I don't think this quite makes sense? Right now, there have been no large breaking changes made to 0.23.1, all the major stuff is new APIs. All the other RL libraries (RLlib, SB3, etc.) have or are about to use a minimum version of the current version of Gym leading up to the 1.0 release. Environments have largely been cleaned up to this end as well, and virtually no reasonable environment code has been broken thus far.

Reply:
this PR should now support both